offline_converter.py

```diff
@@ -160,7 +160,16 @@ def _convert(self, project: QgsProject) -> None:
             xml_elements_to_preserve
         )
         project.readProject.connect(on_original_project_read)
-        project.read(self.backup_filename)
+
+        # if you save the project without QGIS GUI,
+        # the project no longer has `theMapCanvas` canvas
```

Comment:
I am very sorry, I cannot connect this comment to the code below

Reply:
My quick tests showed that saving a project without QGIS GUI `iface` showed there is no longer "theMapCanvas" element. Long time in the past this was fixed via introducing it. It was broken with the last commit on master, because I used the project saved within the QGIS run. https://github.com/opengisch/qfieldcloud-archived/pull/102/files#diff-afdd638b2c3698cf18dc5dfb5ded4369f35d903443219ac4b06f3ad647c4a61b

Reply:
How does this connect to the different project.read calls below?

https://github.com/opengisch/libqfieldsync/pull/28/files#diff-7df75f7cd3e88453bbedd7cee9278cd335db0795e357a5b79beca5db22143867R166-R171

Reply:
`self.backup_filename` is a QGIS projectfile that is saved within the packaging run, so if there is no QGIS GUI, it will save it without `theMapCanvas`. Using this projectfile makes perfect sense when you are using desktop QGIS, because you may have changed the projectfile without saving it. Using the `backup_filename` saves the project temporarily and uses it to package for QField. 

On the cloud side, there is no chance the project would change during the packaging job - there is no user to interact with the project settings. Therefore we can use directly `project.fileName()`. This way the original QGIS projectfile is used that already contains `theMapCanvas`.

Reply:
Can you clarify this in the comment itself (a shorter version is also good enough)?

Reply:
hope it is better now...

Reply:
a lot clearer, thanks